src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -571,7 +534,7 @@ static boolean isCredentialValid(GitHub gitHub) {
                                                     StandardCredentials credentials,
                                                     GitHub github)
             throws IOException {
-        synchronized (githubs) {
+        synchronized (checked) {
```

Comment:
I was concerned about deadlocks at first because we are introducing a new lock, but it looks like this block never does any other synchronization, so I think it should be ok.